test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/TagHelperBlockRewriterTest.cs

```diff
@@ -3612,12 +3612,15 @@ public static TheoryData MinimizedAttributeData_CSharpBlock
 
                     data[1] = buildStatementBlock(() => data[1] as MarkupBlock);
 
-                    var errors = data[2] as RazorError[];
+                    var errors = data[2] as RazorDiagnostic[];
 
                     for (var i = 0; i < errors.Length; i++)
                     {
                         var error = errors[i];
-                        error.Location = SourceLocationTracker.Advance(error.Location, "@{");
+                        var currentErrorLocation = new SourceLocation(error.Span.AbsoluteIndex, error.Span.LineIndex, error.Span.CharacterIndex);
+                        var newErrorLocation = SourceLocationTracker.Advance(currentErrorLocation, "@{");
+                        var newError = new RazorError(error.GetMessage(), newErrorLocation, error.Span.Length);
+                        errors[i] = RazorDiagnostic.Create(newError);
```

Comment:
Some super-hackery here because `RazorError` is mutable and `RazorDiagnostic` is not. I'll clean this up when I get to this.